controllers/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -184,16 +184,22 @@ func (r *NodeNetworkConfigurationPolicyReconciler) Reconcile(ctx context.Context
 			return ctrl.Result{}, err
 		}
 	}
-	defer r.decrementUnavailableNodeCount(instance)
+	defer r.decrementUnavailableNodeCount(policyInstance)
 
 	enactmentConditions.NotifyProgressing()
-	nmstateOutput, err := nmstate.ApplyDesiredState(r.APIClient, instance.Spec.DesiredState)
+	nmstateOutput, err := nmstate.ApplyDesiredState(r.APIClient, policyInstance.Spec.DesiredState)
 	if err != nil {
 		errmsg := fmt.Errorf("error reconciling NodeNetworkConfigurationPolicy at desired state apply: %s, %v", nmstateOutput, err)
 		encodedMessage, _ := encodeMessage(err.Error())
-		strippedErrmsg := fmt.Errorf("error reconciling NodeNetworkConfigurationPolicy at desired state apply: %s,\n %s \n %s", nmstateOutput, formatErrorString(err), encodedMessage)
+		strippedErrmsg := fmt.Errorf("error reconciling NodeNetworkConfigurationPolicy at desired state apply: %s,\n %s", nmstateOutput, formatErrorString(err))
 
 		enactmentConditions.NotifyFailedToConfigure(strippedErrmsg)
+
+		err := r.updateEncodedMessage(policyInstance, encodedMessage)
```

Comment:
Could we also do it here? Assuming `SetFailedToConfigure` handles compression and b64 encoding of course
https://github.com/nmstate/kubernetes-nmstate/blob/a51f39f9152e15ff89fb01ac5e092fcc5bf3a724/pkg/enactmentstatus/conditions/conditions.go#L103